spec/models/user_spec.rb

```diff
@@ -94,6 +94,16 @@
       expect(user).to be_valid
       expect(user.login).to eq "foobar"
     end
+
+    describe "uniqueness" do
+      before { @user = create(:user, login: "jayita") }
+
+      it "can create a user with the same login after deleting" do
+        @user.destroy
+        @user2 = build(:user, login: "jayita")
+        expect(@user2).to be_valid
+      end
+    end
```

Comment:
I would also add a spec to test the main functionality. You want to be sure you didn't break _all_ uniqueness validation by adding the deleted_at scope.

So the same as above but take out the `destroy` part. This should go above the one you've already done. Be sure to have an expectation for the error message as we've done before.